src/codelabs/async-await.md

```diff
@@ -42,6 +42,18 @@ another operation to finish. Here are some common asynchronous operations:
 To perform asynchronous operations in Dart, you can use the `Future` class
 and the `async` and `await` keywords.
 
+Typically, APIs that return `Future` objects
+fall into one of the following categories:
+
+* APIs that are backed by logic running on a different thread. 
+  For example, `File.readAsString` (and most of the other I/O APIs) 
+  call operating system functions from threads on a dedicated thread pool, 
+  rather than from the thread on which Dart code is executed.
+
+* APIs that wait for an event of some kind and perform no work in the meantime.
+  For example, `Future.delayed` waits for time to have elapsed, 
+  and `Stream.first` waits for an event to be posted on the stream.
+
```

Comment:
The distinction seems a little artificial. The "perform no work in the meantime" applies to both categories. Both spin off an asynchronous computation, then wait for it to complete, and does nothing until it completes. The only difference is whether the asynchronous computation is running in the same thread/isolate or another thread. I'm not sure why that matters to the client of the `Future` returning function. They don't need to know, the `Future` is an abstraction over the run-time characteristics of the computation.

If the computation is running in a separate thread, it needs to somehow communicate the result back to the isolate containing the future. When it does that, it usually triggers an asynchronous (top-level, not microtask) event, at which point the future is really just waiting for an asynchronous event in the current isolate.